src/iosrv/srv.c

```diff
@@ -597,7 +597,8 @@ dss_start_one_xstream(hwloc_cpuset_t cpus, int xs_id)
 		 * that creates ULT in DSS_XS_SELF. So ABT_POOL_ACCESS_PRIV
 		 * is fine.
 		 */
-		access = (i == DSS_POOL_SHARE || i == DSS_POOL_REBUILD) ?
+		access = (i == DSS_POOL_SHARE || i == DSS_POOL_REBUILD ||
+			  i == DSS_POOL_URGENT) ?
```

Comment:
"Urgent ult (for dtx_resync) pool should not be PRIV, since dtx_resync can not be private ULT."

seems to me that the dtx_resync only created/scheduled on caller's self XS, why should it be MPSC?

Reply:
dtx_resync needs to send RPC as I see. Anyway this is what I saw in my tests with Fan Yong's patch.

Reply:
ok, we can see why ABT failed later.

Reply:
The dtx_resync() should be scheduled only on the caller's self XS, because it needs to access both the storage and the network. The object rebuild ULT and the container open ULT can trigger dtx_resync() ULT.

Reply:
Hmm, this is not what I saw from my test, you could try rebuild test without this, you can easily see some failure. I will dig further later.

Reply:
I found the reason. The object rebuild will call pool_iter_cb(), that will call dss_ult_create_execute() with DSS_ULT_DTX_RESYNC. The DSS_ULT_DTX_RESYNC will choose DSS_POOL_URGENT pool.

Reply:
Since then I think it is unnecessary to make the dtx_resync to use DSS_POOL_URGENT, instead, just use DSS_POOL_REBUILD may be more suitable.

Reply:
> I found the reason. The object rebuild will call pool_iter_cb(), that will call dss_ult_create_execute() with DSS_ULT_DTX_RESYNC. The DSS_ULT_DTX_RESYNC will choose DSS_POOL_URGENT pool.

In that case, the resync ULT is still created on caller's itself XS.

> Since then I think it is unnecessary to make the dtx_resync to use DSS_POOL_URGENT, instead, just use DSS_POOL_REBUILD may be more suitable.

The original motivation of using special URGET pool for dtx_resync was by the discussion at beijing F2F meeting, that need to ensure the dtx_resync can finish ASAP and avoid the case that when there are lots of IO tasks possibly cause the dtx_resync be delayed long time.